xapian-bindings/python3/Makefile.am

```diff
@@ -172,3 +174,109 @@ $(sphinxdocs): xapian/__init__.py xapian/_xapian$(PYTHON3_SO) docs/conf.py $(RST
 	PYTHONPATH=..:$$PYTHONPATH $(OSX_SIP_HACK_ENV) $(PYTHON3) \
 		$(srcdir)/sphinx-build \
 		-b html -d doctrees -c docs $(srcdir)/docs docs/html
+
+
+if BUILD_LETOR_BINDINGS
+TESTS += pythonletortest.py
+BUILT_SOURCES += xapianletor.py \
+	xapianletor_wrap.cc \
+	xapianletor_wrap.h
+
+EXTRA_DIST += pythonletor.i \
+	exceptletor.i \
+	generate-letor-python-exceptions \
+	test_xapian_letor_star.py \
+	pythonletortest.py \
+	xapianletor.py \
+	xapianletor_wrap.cc \
+	xapianletor_wrap.h
```

Comment:
If I take the distribution tarball `xapian-bindings-1.5.0.tar.xz` then unpack it, generate a new distribution tarball, and compare the two tarballs:

```
tar xf xapian-bindings-1.5.0.tar.xz
cd xapian-bindings-1.5.0
# Need to specify XAPIAN_CONFIG here only because I haven't installed xapian-core-1.5.0
./configure XAPIAN_CONFIG=~/git/xapian/xapian-core/xapian-config
make dist
cd ..
tardiff xapian-bindings-1.5.0.tar.xz xapian-bindings-1.5.0/xapian-bindings-1.5.0.tar.xz
```

then the `tardiff` output shows that the following files are missing:

```
- python/doccomments.i
- python3/doccomments.i
- python3/exceptletor.i
- python3/generate-letor-python-exceptions
- python3/pythonletor.i
- python3/pythonletortest.py
- python3/test_xapian_letor_star.py
- python3/xapianletor.py
- python3/xapianletor_wrap.h
```

The first two are an existing bug here (which I've just pushed 5198109aacc7cfa2edfb611039fe216003eb37c5 to fix) but the rest indicate that this `EXTRA_DIST += ...` should be outside the conditional.

Also, this highlights that `xapianletor_wrap.cc` and `xapianletor_wrap.h` don't need to be listed here (note that `xapian_wrap.cc` and `xapian_wrap.h` aren't listed in `EXTRA_DIST`).

Reply:
I think you forgot to configure with --enable-letor. 
> Also, this highlights that `xapianletor_wrap.cc` and `xapianletor_wrap.h` don't need to be listed here (note that `xapian_wrap.cc` and `xapian_wrap.h` aren't listed in `EXTRA_DIST`).
>
I think `xapian_wrap.h` and `xapian_wrap.cc` are listed in `EXTRA_DIST` as `BUILT_SOURCES`.

Reply:
Right, but they aren't listed in `EXTRA_DIST` explicitly like the letor ones currently are.

(And I did check that removing the letor ones works before commenting...)

Reply:
Sorry, I missed this part of the comment before:

> I think you forgot to configure with --enable-letor.

No, that was a deliberate omission.

My point here is that what `make dist` produces should **not** depend on which configure options were or were not specified - if it does, there's a trap for someone to build an incomplete distribution.  Repacking a distribution like this shouldn't be a lossy process.

Reply:
fixed in a fixup commit.